Add support for Tag entity with DTOs and tests

marierinden opened the pull request:
### Description

This pull request introduces support for the `Tag` entity in the SDK, including the addition of DTO classes and resources to manage Tag-related operations: creation, retrieval, updating, and deletion. The following components are introduced:

- `TagRequest`: DTO for creating and updating tags.
- `TagResponse`: DTO for retrieving individual tag details.
- `TagCollectionResponse`: DTO for handling collections of tags.
- `TagResource`: Resource class for managing API interactions for tags.

Additionally, unit tests have been added for `TagRequest`, `TagResponse`, and `TagCollectionResponse` to validate their structure, data handling, and expected behaviors. These enhancements ensure reliable functionality for tag operations.

marierinden committed:
Add support for Tag entity in SDK

Introduced DTOs and resource classes for managing Tag operations, including creation, retrieval, updating, and deletion. This addition includes `TagRequest`, `TagResponse`, `TagCollectionResponse`, and `TagResource` to handle tag-related API interactions.

marierinden committed:
Add unit tests for Tag DTO request and response classes

Introduce unit tests for `TagRequest`, `TagResponse`, and `TagCollectionResponse`. These tests validate the structure, data handling, and expected behaviors of the respective DTOs to ensure reliable functionality.

fredrik-tveraaen-apiera commented on src/DTO/Request/Tag/TagRequest.php:
```diff
@@ -0,0 +1,52 @@
+final readonly class TagRequest implements RequestInterface
+{
+    public function __construct(
+        #[RequestField('name')]
+        private string $name,
+        #[RequestField('store')]
+        private string $store,
```
Change to nullable properties

marierinden committed:
Make TagRequest fields nullable

Updated the `name` and `store` fields in `TagRequest` to be nullable and adjusted their getters accordingly. This allows handling cases where these fields might not be provided when creating or processing a request.

marierinden committed:
Make `name` and `store` nullable in TagResponse

Updated the `TagResponse` DTO to allow `name` and `store` fields to be nullable, aligning with potential API responses. Adjusted related unit tests to reflect these changes and ensure proper handling of null values.

fredrik-tveraaen-apiera committed:
Refactor Tag DTOs to enforce non-nullable fields

Updated `name` and `store` fields in `TagResponse` to be non-nullable, ensuring stronger data integrity. Adjusted tests and corresponding request/response data to reflect these changes. Removed unused `store` field from `TagRequest` serialization.

fredrik-tveraaen-apiera committed:
Add support for Tag resource and response mapping

Introduced TagResponse and TagCollectionResponse mappings to `LdType`. Updated `ApieraSdk` to include a `tag()` method returning a `TagResource`. This change enables handling tags as a distinct resource in the SDK.

fredrik-tveraaen-apiera committed:
Add integration tests for Tag resource CRUD operations

This commit introduces tests for Create, Read, Update, and Delete operations related to the Tag resource. These tests validate proper request and response handling, as well as resource-specific behaviours in a store-scoped context.

fredrik-tveraaen-apiera committed:
Merge release/0.3.0

fredrik-tveraaen-apiera reviewed (approved):
:100:
